src/pymor/models/basic.py

```diff
@@ -89,12 +89,14 @@ def _compute_solution(self, mu=None, **kwargs):
 
     @property
     def dual(self):
-        if not hasattr(self, '_dual'):
-            assert self.output_functional is not None
-            assert self.output_functional.linear
-            # TODO: assert that the operator is symmetric
+        assert self.output_functional is not None
+        assert self.output_functional.linear
```

Comment:
Maybe these assertions only need to happen when the members are updated?

Reply:
Yes, sure.